erpnext/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -321,13 +321,25 @@ def get_working_days_details(self, joining_date=None, relieving_date=None, lwp=N
 			self.payment_days = 0
 
 	def get_unmarked_days(self):
-		marked_days = frappe.get_all("Attendance", filters = {
-					"attendance_date": ["between", [self.start_date, self.end_date]],
-					"employee": self.employee,
-					"docstatus": 1
-				}, fields = ["COUNT(*) as marked_days"])[0].marked_days
+		unmarked_days = self.total_working_days
+		joining_date, relieving_date = frappe.get_cached_value("Employee", self.employee,
+			["date_of_joining", "relieving_date"])
+
+		if joining_date:
+			if getdate(self.start_date) < joining_date <= getdate(self.end_date):
+				unmarked_days -= date_diff(joining_date, self.start_date)
+
+		if relieving_date:
+			if getdate(self.start_date) <= relieving_date < getdate(self.end_date):
+				unmarked_days -= date_diff(relieving_date, self.end_date)
```

Comment:
```suggestion
				unmarked_days -= date_diff(self.end_date, relieving_date)
```

`date_diff` calculates difference as `end_date - start_date`:

https://github.com/frappe/frappe/blob/acac60d061bbb1ee4957ca613c92e1526098aa3b/frappe/utils/data.py#L166-L167

So in this case since `end_date` would be after `relieving_date` its the end date

Reply:
Thanks for the fix. 

- Can you all add separate tests for both these cases: Date of Joining and Relieving Date. 
- Also, the description is fairly understandable, but can you update the description with a screenshot depicting the before (negative days calculated) and after (payment days calculation working fine).

Reply:
Sure, will do.